objgen.py

```diff
@@ -0,0 +1,76 @@
+import json
+import os
+from collections import defaultdict
+from ipaddress import IPv4Network, IPv6Network
+from typing import List
+
+
+class AnnouncementGenerator:
+    prefix: IPv4Network or IPv6Network
+    muxes: List[str]
+    details: List[dict]
+
+    def __init__(
+        self,
+        prefix,
+        muxes,
+        origin: int = None,
+        peers: List[int] = None,
+        prepend: List[int] = None,
+    ):
+        self.prefix = prefix
+        self.muxes = muxes
+        self.details = list()
+
+        self.add_details(muxes, origin, peers, prepend)
+
+    def add_details(
+        self,
+        muxes,
+        origin: int = None,
+        peers: List[int] = None,
+        prepend: List[int] = None,
+    ):
+        dictdetails = dict()
+        dictdetails["muxes"] = muxes
+        if origin:
+            dictdetails["origin"] = origin
+        if peers:
+            dictdetails["peers"] = peers
+        if prepend:
+            dictdetails["prepend"] = prepend
+        self.details.append(dictdetails)
```

Comment:
It seems like this whole function can be turned into a one-liner by using `**kwargs`.